app/questionnaire/location.py

```diff
@@ -0,0 +1,71 @@
+from flask import url_for
+
+
+class Location(object):
+
+    def __init__(self, group_id, group_instance, block_id):
+
+        self.group_id = group_id
+        self.group_instance = group_instance
+        self.block_id = block_id
+
+    def __eq__(self, other):
+        """
+        Check to see if two locations are equal.
+        Two answers are considered to be equal if their dictionary representations equal one another.
+
+        :param other: An answer to compare
+        :return: True if both answers match, otherwise False.
+        """
+        return self.__dict__ == other.__dict__
```

Comment:
From [Fluent Python](https://www.amazon.co.uk/Fluent-Python-Luciano-Ramalho/dp/1491946008)

How Special Methods Are Used
The first thing to know about special methods is that they are meant to be called by the Python interpreter, and not by you. You don’t write `my_object.__len__()`. You write len(my_object) and, if my_object is an instance of a user-defined class, then Python calls the `__len__` instance method you implemented.